src/components/inspectors/TimerExpression.vue

```diff
@@ -0,0 +1,381 @@
+    for (let i = 0; i < 48; i++) {
+      hours.push(date.format('HH:mm'));
+      date.add(30, 'minutes');
+    }
+    return {
+      today: moment().format('YYYY-MM-DD'),
+      hours,
+      weekdays: [
+        //  ISO week date weekday number, from 1 through 7,
+        //  beginning with Monday and ending with Sunday.
+        {
+          day: 7,
+          initial: 'S',
+          selected: false,
+        },
+        {
+          day: 1,
+          initial: 'M',
+          selected: false,
+        },
+        {
+          day: 2,
+          initial: 'T',
+          selected: false,
+        },
+        {
+          day: 3,
+          initial: 'W',
+          selected: false,
+        },
+        {
+          day: 4,
+          initial: 'T',
+          selected: false,
+        },
+        {
+          day: 5,
+          initial: 'F',
+          selected: false,
+        },
+        {
+          day: 6,
+          initial: 'S',
+          selected: false,
+        },
+      ],
+      startDate: new Date(),
+      startTime: '00:00',
+      repeat: '1',
+      periodicity: 'week',
+      ends: 'never',
+      endDate: new Date(),
+      times: '1',
+    };
+  },
+  computed: {
+    expression() {
+      this.startDate;
+      this.startTime;
+      this.repeat;
+      this.periodicity;
+      this.selectedWeekdays;
+      this.ends;
+      this.endDate;
+      this.times;
+      const expression = this.makeTimerConfig();
+      return expression;
+    },
+    selectedWeekdays() {
+      const selected = [];
+      this.weekdays.forEach(weekday => {
+        weekday.selected ? selected.push(weekday.day) : null;
+      });
+      return selected;
+    },
+    sameDay() {
+      return this.selectedWeekdays.length === 1 && this.weekdays[this.startDate.getDay()].selected;
+    },
+  },
+  watch: {
+    value(value) {
+      this.parseTimerConfig(value);
+    },
+  },
+  methods: {
+    currentDaySelected(day) {
+      return this.startDate.getDay() === (day.day % 7);
+    },
+    updateStartDate(date) {
+      this.startDate = date;
+      this.update();
+    },
+    updateEndDate(date) {
+      this.endDate = date;
+      this.update();
+    },
+    update() {
+      this.$emit('input', this.expression);
+    },
+    parseDateExpression(exp) {
+      const date = new Date(exp);
+      if (isNaN(date.getTime())) {
+        throw 'Invalid Date';
+      }
+      return moment(date);
+    },
+    parseTimerConfig(value) {
+      this.resetTimerExpression();
+      if (!value) {
+        return;
+      }
+      try {
+        let date, hasStartDate = false;
+        const expression = value.split('|');
+        expression.forEach(exp => {
+          if (exp.substr(0, 1) !== 'R') {
+            date = this.parseDateExpression(exp);
+            this.startDate = date.toDate();
+            this.startTime = date.format('HH:mm');
+          } else {
+            let match = exp.match(/R(\d*)\/([^/]+)\/P(\d+)(\w)(?:\/([^/]+))?/);
```

Comment:
Could you add a comment explaining what this pattern does ?

Reply:
Added code comments explaning the pattern